.config/nextest.toml

```
@@ -6,9 +6,22 @@ fail-fast = true
 status-level = "pass"
 
 # --- All Tests profile ---
-# Runs all tests except 'check_no_git_dependencies'
+# CI-friendly test selection.
+#
+# Runs fast unit/integration tests and explicitly excludes:
+# - 'check_no_git_dependencies' (it has its own profile)
+# - stateful or long-running tests that require a local cache dir or external
+#   services (lightwalletd RPC/GRPC, full sync, checkpoint generation, mining
+#   RPCs, etc.).
+#
+# Notes:
+# - CI now provides a default cache directory, so stateful tests are no longer
+#   auto-skipped when the cache dir is undefined. To keep this profile fast and
+#   deterministic, we exclude those tests here and offer dedicated profiles
+#   below for running them when needed.
+# TODO: We need a better test architecture to run all non-stateful
 [profile.all-tests]
-default-filter = 'not test(check_no_git_dependencies)'
+default-filter = "not test(check_no_git_dependencies) and not test(=fully_synced_rpc_z_getsubtreesbyindex_snapshot_test) and not test(=lwd_rpc_test) and not test(=lwd_rpc_send_tx) and not test(=lwd_grpc_wallet) and not test(=lwd_integration) and not test(=lwd_sync_full) and not test(=lwd_sync_update) and not test(=lightwalletd_test_suite) and not test(=rpc_get_block_template) and not test(=rpc_submit_block) and not test(=get_peer_info) and not test(~generate_checkpoints_) and not test(=trusted_chain_sync_handles_forks_correctly) and not test(=sync_update_mainnet) and not test(=trusted_chain_sync_handles_forks_correctly) and not test(=activate_mempool_mainnet) and not test(=downgrade_state_format) and not test(=invalidate_and_reconsider_block) and not test(=nu6_funding_streams_and_coinbase_balance)"
```

I think we should be able to get some of these working

```suggestion
default-filter = "not test(check_no_git_dependencies) and not test(=fully_synced_rpc_z_getsubtreesbyindex_snapshot_test) and not test(=lwd_rpc_test) and not test(=lwd_rpc_send_tx) and not test(=lwd_grpc_wallet) and not test(=lwd_integration) and not test(=lwd_sync_full) and not test(=lwd_sync_update) and not test(=lightwalletd_test_suite) and not test(=rpc_get_block_template) and not test(=rpc_submit_block) and not test(=get_peer_info) and not test(~generate_checkpoints_) and not test(=sync_update_mainnet) and not test(=activate_mempool_mainnet)"
```

---

I merged your PR, so this should be good now